spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -575,13 +577,22 @@ else if (pool.getType() == FIXED) {
 					X509Certificate[] trustedX509Certificates = ssl
 							.getTrustedX509CertificatesForTrustManager();
 					if (trustedX509Certificates.length > 0) {
-						sslContextBuilder.trustManager(trustedX509Certificates);
+						sslContextBuilder = sslContextBuilder
+								.trustManager(trustedX509Certificates);
 					}
 					else if (ssl.isUseInsecureTrustManager()) {
-						sslContextBuilder
+						sslContextBuilder = sslContextBuilder
 								.trustManager(InsecureTrustManagerFactory.INSTANCE);
 					}
 
+					if (ssl.getKeyCertChain() != null
+							&& ssl.getKeyCertChain().length() > 0 && ssl.getKey() != null
+							&& ssl.getKey().length() > 0) {
+						sslContextBuilder = sslContextBuilder.keyManager(
+								new File(ssl.getKeyCertChain()), new File(ssl.getKey()),
```

Comment:
`File` is probably a bad idea here since they might come from a key store, or not be in the file system. Is there an alternative API?